src/main/kotlin/io/github/nomisrev/service/ArticleService.kt

```diff
@@ -32,7 +32,9 @@ fun articleService(
     override suspend fun createArticle(input: CreateArticle): Either<DomainError, Article> =
       either {
         val slug =
-          slugGenerator.generateSlug(input.title) { slug -> articlePersistence.exists(slug) }.bind()
+          slugGenerator
+            .generateSlug(input.title) { slug -> articlePersistence.exists(slug).not() }
+            .bind()
```

Comment:
Using the 'exist' function to implement 'unique' requires that the value does not exist in the database, so the logic was inverted to achieve the expected behavior.

Reply:
This is was missing a test in that case, but doesn't need to be handled in this PR. Will add an issue for it.